test/e2e/apimachinery/crd_conversion_webhook.go

```diff
@@ -0,0 +1,295 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"time"
+
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	utilversion "k8s.io/kubernetes/pkg/util/version"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	_ "github.com/stretchr/testify/assert"
+)
+
+const (
+	secretCRDName      = "sample-custom-resource-conversion-webhook-secret"
+	deploymentCRDName  = "sample-crd-conversion-webhook-deployment"
+	serviceCRDName     = "e2e-test-crd-conversion-webhook"
+	roleBindingCRDName = "crd-conversion-webhook-auth-reader"
+)
+
+// TODO(mehdy): this should be 1.12.0
+var serverCRDConversionWebhookVersion = utilversion.MustParseSemantic("v1.11.0")
+
+var _ = SIGDescribe("CustomResourceConversionWebhook [Feature:CustomResourceWebhookConversion]", func() {
+	var context *certContext
+	f := framework.NewDefaultFramework("crd-webhook")
+
+	var client clientset.Interface
+
+	BeforeEach(func() {
+		client = f.ClientSet
+
+		// Make sure the relevant provider supports admission webhook
+		framework.SkipUnlessServerVersionGTE(serverCRDConversionWebhookVersion, f.ClientSet.Discovery())
+		framework.SkipUnlessProviderIs("gce", "gke", "local")
```

Comment:
why?

Reply:
I also have this question. Got this from admission webhooks. @caesarxuchao do you know why?

Reply:
aws doesn't support webhook back then, not sure if it's necessary anymore.